### PR TITLE
Added Support for AWS EC2 Reserved Instances

### PR DESCRIPTION
**Pull request by kedarghule**

### CODE REVIEW

**Comment by kedarghule** [comment]
Don't merge Cloudanix branch in this xD

File: cartography/intel/aws/ec2/reserved_instances.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+from cartography.util import run_cleanup_job
+from cartography.util import timeit
+
+logger = logging.getLogger(__name__)
+
+
+@timeit
+@aws_handle_regions
+def get_reserved_instances(boto3_session: boto3.session.Session, region: str) -> List[Dict]:
+    client = boto3_session.client('ec2', region_name=region, config=get_botocore_config())
```
Comment (mpurusottamc) [inline cartography/intel/aws/ec2/reserved_instances.py:20]: let's get rid of config parameter in this client creation. Ultimately the goal is to get rid of these config assignments. Instead, users should use environment variables for these settings.

We should do the same in other ec2 resources as well.

**Review by mpurusottamc — CHANGES_REQUESTED** [review]
Only 1 change. rest looks good.

File: cartography/intel/aws/ec2/reserved_instances.py (hunk 1 above)
Comment (kedarghule) [reply cartography/intel/aws/ec2/reserved_instances.py:20]: Done

**Review by mpurusottamc — APPROVED** [review]

**Pull request merged**